src/Api/Domain.php

```diff
@@ -299,4 +303,80 @@ public function verify(string $domain)
 
         return $this->hydrateResponse($response, VerifyResponse::class);
     }
+
+    /**
+     * Returns a domain tracking settings.
+     *
+     * @param string $domain name of the domain
+     *
+     * @return TrackingResponse|array|ResponseInterface
+     */
+    public function tracking(string $domain)
+    {
+        Assert::stringNotEmpty($domain);
+
+        $response = $this->httpGet(sprintf('/v3/domains/%s/tracking', $domain));
+
+        return $this->hydrateResponse($response, TrackingResponse::class);
+    }
+
+    /**
+     * Updates a domain click tracking settings.
+     *
+     * @param string $domain name of the domain
+     *
+     * @param bool $active
+     * @return UpdateClickTrackingResponse|array|ResponseInterface
+     */
+    public function updateClickTracking(string $domain, bool $active)
+    {
+        $params = [
+            'active' => $active ? 'true' : 'false',
+        ];
+
+        $response = $this->httpPut(sprintf('/v3/domains/%s/tracking/click', $domain), $params);
+
+        return $this->hydrateResponse($response, UpdateClickTrackingResponse::class);
+    }
+
+    /**
+     * Updates a domain open tracking settings.
+     *
+     * @param string $domain name of the domain
+     * @param bool $active
+     *
+     * @return UpdateOpenTrackingResponse|array|ResponseInterface
+     */
+    public function updateOpenTracking(string $domain, bool $active)
+    {
+        $params = [
+            'active' => $active ? 'true' : 'false',
+        ];
+
+        $response = $this->httpPut(sprintf('/v3/domains/%s/tracking/open', $domain), $params);
+
+        return $this->hydrateResponse($response, UpdateOpenTrackingResponse::class);
+    }
+
+    /**
+     * Updates a domain unsubscribe tracking settings.
+     *
+     * @param string $domain name of the domain
+     * @param bool $active
+     * @param string $htmlFooter
+     * @param string $textFooter
+     * @return UpdateUnsubscribeTrackingResponse|array|ResponseInterface
+     */
+    public function updateUnsubscribeTracking(string $domain, bool $active, string $htmlFooter, string $textFooter)
+    {
+        $params = [
+            'active' => $active ? 'true' : 'false',
+            'html_footer' => $htmlFooter,
+            'text_footer' => $textFooter,
```

Comment:
According to the Mailgun Documentation, it's not clear for me if the values for the `active` property must be a `bool` or a `string`. However, it seems like at least the `text_footer` might be optional, given the API documentation states that it can be generated automatically by the Mailgun system.

Can you fix this implementation, please, making sure we are adding the parameters just if they are strictly mandatory, and not adding them when they are optional?

Please don't forget adding the expected `Asset::...` before setting the parameters. It would be great if you also add additional unit tests to validate the exceptions thrown by the `Asset` methods.

https://documentation.mailgun.com/en/latest/api-domains.html#domains

![image](https://user-images.githubusercontent.com/7809429/95641976-93c16380-0a9d-11eb-912f-e282f5820df4.png)

Reply:
All done, except the optional text_footer, because seems like its mandatory, because API returns an error:
```
{
  "message": "Missing parameter 'text_footer'"
}
```
when I'm not passing text_footer or passing it empty